### PR TITLE
Fix file links

### PR DESCRIPTION
**Pull request by orizerah**

### CODE REVIEW

File: environments/azure/environment.go [hunk 1]
```diff
@@ -165,22 +165,41 @@ func (e environment) GetBuildLink() string {
...
-		return fmt.Sprintf("%s_git/%s?path=%s&version=GB%s&line=%d&lineEnd=%d&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents",
-			os.Getenv(endpointURLEnv),
-			os.Getenv(repositoryNameEnv),
-			strings.Replace(filePath, "/", "%2F", -1),
-			branchName,
-			lineNumber,
-			lineNumber+1,
+func (e environment) GetFileLineLink(filePath string, branch string, commit string, startLine int, endLine int) string {
+	return GetFileLink(
+		fmt.Sprintf("%s_git/%s", os.Getenv(endpointURLEnv), os.Getenv(repositoryNameEnv)),
```
Comment (davidsalame1) [inline environments/azure/environment.go:170]: What is the problem with line 0?

File: global/filelink.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	"github.com/argonsecurity/go-environments/enums"
+	"github.com/argonsecurity/go-environments/environments/azure"
+	"github.com/argonsecurity/go-environments/environments/bitbucket"
+	"github.com/argonsecurity/go-environments/environments/github"
+	"github.com/argonsecurity/go-environments/environments/gitlab"
+)
+
+type GetFileLinkFunc func(string, string, string, string, int, int) string
+
+func GetFileLink(source enums.Source, repositoryURL string, filename string, branch string, commit string, startLine int, endLine int) string {
```
Comment (davidsalame1) [inline global/filelink.go:13]: Why do you need this file?
You can add "GetFileLink" on the base environment.go like we did for getConfiguration, no? @orizerah

File: environments/azure/environment.go (hunk 1 above)
Comment (orizerah) [reply environments/azure/environment.go:170]: Why is there a problem with line 0?

File: environments/azure/environment.go [hunk 3]
```diff
@@ -165,22 +165,41 @@ func (e environment) GetBuildLink() string {
...
+		filePath,
+		branch,
+		commit,
+		startLine,
+		endLine,
+	)
+}
+
+func GetFileLink(repositoryURL string, filename string, branch string, commit string, startLine, endLine int) string {
```
Comment (VaismanLior) [inline environments/azure/environment.go:179]: I would rename this func.
This looks like `GetFileLineLink` is a link with line numbers and `GetFileLink` is without line numbers.
I would rename it to `getAzureLink` and make it private(lowercase)
Same for all the providers

File: environments.go [hunk 4]
```diff
@@ -36,7 +38,7 @@ type Environment interface {
 	GetStepLink() string
 
 	// GetFileLineLink get a link to a file line
-	GetFileLineLink(filename string, ref string, line int) string
+	GetFileLineLink(filename string, ref string, commitId string, startLine int, endLine int) string
```
Comment (davidsalame1) [inline environments.go:42]:
```suggestion
	GetFileLineLink(filename string, ref string, commit string, startLine int, endLine int) string

```

File: environments/localhost/environment.go [hunk 5]
```diff
@@ -74,7 +74,7 @@ func (e environment) GetBuildLink() string {
 	return "localhost"
 }
 
-func (e environment) GetFileLineLink(filename string, ref string, line int) string {
+func (e environment) GetFileLineLink(filename string, ref string, commitId string, startLine int, endLine int) string {
 	return "localhost"
```
Comment (davidsalame1) [inline environments/localhost/environment.go:78]:
```suggestion
func (e environment) GetFileLineLink(filename string, ref string, commit string, startLine int, endLine int) string {
	return "localhost"
```

**Review by davidsalame1 — APPROVED** [review]

**Review by davidsalame1 — DISMISSED** [review]

**Review by davidsalame1 — APPROVED** [review]

**Pull request merged**